pkg/datapath/iptables/iptables.go

```diff
@@ -694,6 +707,68 @@ func (m *IptablesManager) iptProxyRules(cmd string, proxyPort uint16, ingress bo
 	return nil
 }
 
+func noTrackRules(prog string, cmd string, IP string, port *lb.L4Addr, ingress bool) error {
+	protocol := strings.ToLower(port.Protocol)
+	p := strconv.FormatUint(uint64(port.Port), 10)
+	if ingress {
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "raw", cmd, "PREROUTING", "-p", protocol, "-d", IP, "--dport", p, "-j", "NOTRACK"}, false); err != nil {
+			return err
+		}
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "filter", cmd, "INPUT", "-p", protocol, "-d", IP, "--dport", p, "-j", "ACCEPT"}, false); err != nil {
+			return err
+		}
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "raw", cmd, "OUTPUT", "-p", protocol, "-d", IP, "--dport", p, "-j", "NOTRACK"}, false); err != nil {
+			return err
+		}
+	} else {
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "raw", cmd, "OUTPUT", "-p", protocol, "-s", IP, "--sport", p, "-j", "NOTRACK"}, false); err != nil {
+			return err
+		}
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "filter", cmd, "OUTPUT", "-p", protocol, "-s", IP, "--sport", p, "-j", "ACCEPT"}, false); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func InstallNoTrackRules(IP string, port uint16, ipv6 bool) error {
+	prog := "iptables"
+	if ipv6 {
+		prog = "ip6tables"
+	}
+	ports := noTrackPorts(port)
+	for _, p := range ports {
+		if err := noTrackRules(prog, "-A", IP, p, true); err != nil {
+			log.WithError(err).Warnf("Unable to install ingress NOTRACK rules for %s:%d with %s", IP, port, prog)
```

Comment:
Please use structured logging, ie `log.WithFields(...).WithError(err).Warn("Unable to install ingress NOTRACK rules")`. This way, when the fields are pushed to a log collector that is familiar with structured fields, it can correlate errors with the same message or allow you to search log messages with a common field.

Reply:
Done.